### ISSUE TITLE
google_checks.xml: incorrect warnings for empty try with resources blocks

### ISSUE THREAD

**Issue by cushon** [checkstyle/checkstyle#10608]
```bash
$ java -jar checkstyle-8.45.1-all.jar -c /google_checks.xml T.java
...
[WARN] T.java:3:31: Empty try block. [EmptyBlock]
[WARN] T.java:3:31: WhitespaceAround: '{' is not followed by whitespace. Empty blocks may only be represented as {} when not part of a multi-block statement (4.1.3) [WhitespaceAround]
[WARN] T.java:3:32: WhitespaceAround: '}' is not preceded with whitespace. [WhitespaceAround]
```

```java
class T {
  {
    try (Foo foo = new Foo()) {}
  }

  static class Foo implements AutoCloseable {
    @Override
    public void close() {}
  }
}
```

The Google Java Style Guide permits the above code, it should not result in a warning with `google_checks.xml`.

From https://google.github.io/styleguide/javaguide.html#s4.1.3-braces-empty-blocks

> An empty block or block-like construct ... may be closed immediately after it is opened, with no characters or line break in between (`{}`), unless it is part of a multi-block statement (one that directly contains multiple blocks: `if`/`else` or `try`/`catch`/`finally`).

The part about `try`/`catch`/`finally` doesn't apply to try with resources block without a `catch` or `finally` block.

**Comment by cushon** [comment]
Similarly, this shouldn't result in a warning either:

```java
class T {
  void f(boolean b) {
    if (b) {}
  }
}
```

```
[WARN] T.java:3:12: Empty if block. [EmptyBlock]
[WARN] T.java:3:12: WhitespaceAround: '{' is not followed by whitespace. Empty blocks may only be represented as {} when not part of a multi-block statement (4.1.3) [WhitespaceAround]
[WARN] T.java:3:13: WhitespaceAround: '}' is not preceded with whitespace. [WhitespaceAround]
```

**Comment by romani** [comment]
@cushon , do you know if this is regression in recent releases ? Or it was like this from the beginning

**Comment by Z4ID-Reaper** [comment]
@romani @Zopsss 
hi , apologies for tagging you guys at so many issues , there are total to 74 in this project 😄 

This issue was very long ago opened , now its solved in the latest versions of checkstyle, as shown below.
hence it seems we should close it .

### config

i have used the google_style.xml one from here 👉 https://checkstyle.org/cmdline.html#Download_and_Run

---
 ### Test.java

```
class T {
  {
    try (Foo foo = new Foo()) {}
  }

  static class Foo implements AutoCloseable {
    @Override
    public void close() {}
  }
}
```

---

### cli output

```
D:\zaid\zaid\codes\asasas> java -jar checkstyle-10.23.1-all.jar -c /google_checks.xml Test.java
Starting audit...
Audit done.
```

---

**Comment by cushon** [comment]
It appears to have been fixed in https://github.com/checkstyle/checkstyle/releases/tag/checkstyle-10.18.2, the release notes for that release mention EmptyBlock.

```
$ java -jar checkstyle-10.18.1-all.jar -c /google_checks.xml T.java
Starting audit...
[WARN] T.java:3:31: Empty try block. [EmptyBlock]
[WARN] T.java:3:31: WhitespaceAround: '{' is not followed by whitespace. Empty blocks                may only be represented as {} when not part of a multi-block statement (4.1.3) [WhitespaceAround]
[WARN] T.java:3:32: WhitespaceAround: '}' is not preceded with whitespace. [WhitespaceAround]
Audit done.

$ java -jar checkstyle-10.18.2-all.jar -c /google_checks.xml T.java
Starting audit...
Audit done.
```

**Comment by romani** [comment]
Thanks a lot for checking this.
Closing.
